modules/config/home-cursor.nix

```diff
@@ -52,6 +52,27 @@ let
       escapeShellArg cfg.x11.defaultCursor
     }";
 
+  defaultIndexThemePackage = pkgs.writeTextFile {
+    name = "index.theme";
+    destination = "/share/icons/default/index.theme";
+    # Set name in icons theme, for compatibility with AwesomeWM etc. See:
+    # https://github.com/nix-community/home-manager/issues/2081
+    # https://wiki.archlinux.org/title/Cursor_themes#XDG_specification
+    text = ''
+      [Icon Theme]
+      Name=Default
+      Comment=Default Cursor Theme
+      Inherits=${cfg.name}
+    '';
+  };
+
+  iconMappings = let
+    knownFiles = filterAttrs (_: type: type != "unknown")
+      (builtins.readDir "${cfg.package}/share/icons");
```

Comment:
Unfortunately, this line introduced an IFD into home-manager, causing eval errors in my CI system that does not allow for IFD.

Is there an existing policy on IFD in home-manager? In nixpkgs for instance IFD is not allowed.

I think we should at least retain the option of simply recursively linking the cursor dirs instead of relying on IFD, to account for situations where IFD is not allowed or desirable.

Any opinions?

Reply:
My apologies, I am not too familiar with IFD and thus overlooked the IFD being introduced. I think in this case the IFD is not strictly necessary and can be avoided by symlinking only the directory under the icon package matching `config.home.pointerCursor.name`. This should not result in any functional difference in behavior from the status quo as the module is only meant to generate configurations for the specified cursor. I have implementing the above change in https://github.com/nix-community/home-manager/pull/4369.

Reply:
No need to apologise! IFDs are easily missed.

Thanks for the quick fix!